### ISSUE TITLE
Reading SnpEffects.mcmcsamples.bin

### ISSUE THREAD

**Issue by avnostaeva** [zhilizheng/SBayesRC#31]
Dear Zhili,

Thanks for your work!

I want to try a recently proposed method (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8758557/) for calculating polygenic risks based on the use of posterior samples of the genetic effects. If I understand correctly, the values ​​I need are in the file SnpEffects.mcmcsamples.bin. I use the gctb tool with --sbayes R option. 

1) I would like to clarify what information is contained in this file?
2) How to read this file to extract it? I tried the option --mcmc-samples, but it doesn't work.
3) In the file *snpRes, how are the values ​​in column SE calculated?

Thank you in advance! If I have contacted the wrong address, please write where it is better to contact.

Best,
Arina

**Comment by avnostaeva** [comment]
Hello!

Thank you! Especially for the clarification about scaling.

Do I understand correctly that in all implementations of the algorithm (gctb --sbayes R, gctb --sbayes RC and SBayesRC::sbayesrc), input summary statistics can be unscaled? Am I right that scaling is done from the SE and N values (from sum.stat) ​​at the beginning of the algorithm (marginal beta multiplying by sqrt(1/(N*SE^2))), and at the end of the algorithm, the SNP weights are returned to the normal scale by dividing by the same scale-number?

Best,
Arina

**Comment by zhilizheng** [comment]
Hi @avnostaeva ,

You are right.  This was exactly what we did in the code and described in our paper. 

Regards,
Zhili

**Comment by harryyiheyang** [comment]
Hi @zhilizheng 

I am currently working on a project where I need to obtain the standard error (SE) of BETA for some specific purposes as the PIP alone is not sufficient for my needs.

I have tried setting bOutBeta=T and bOutDetail=T, but I encountered an issue with the file size. The output file size for each MCMC iteration is approximately 91MB, resulting in a total size of about 300GB for 3000 iterations. This size is too large for practical use.

Could you please advise if there is a more efficient way to obtain the SE of BETA during the MCMC process? Any guidance or suggestions you could provide would be greatly appreciated.

Thank you for your time and for providing such an excellent tool.

**Comment by zhilizheng** [comment]
@harryyiheyang,

Oh, that's horrible.   I will add the feature in next week if I can manage the time (SE of beta).     

I will also create a new way to save the BETA in a very efficient way, that can also directly used in generating the PRS.   That may be useful for @avnostaeva 

Regards,
Zhili

**Comment by zhilizheng** [comment]
We only save the 2000 iterations after the burn-in.  So the storage would be 200GB or so.   I will cut this to 60GB, and also flag to specify the thin flag, if in a step of 5, then the storage would be 12GB. 

@harryyiheyang If you are in a hurry, you can change the chain length to a smaller value, e.g.,  niter=2000.  However, it will decrease the prediction accuracy for some traits by 2%. 

Regards,
Zhili

**Comment by harryyiheyang** [comment]
@zhilizheng 
Thank you! I would try your suggestion using niter and thin flag = 5.
It will be great that you could manage a time to add the feature (SE of beta).

**Comment by zhilizheng** [comment]
Hi @harryyiheyang ,

Sorry for the delay. I have a busy two week.  I'm updating this weekends.  Hope I can finish it in the early next week. 

Regards,
Zhili

**Comment by zhilizheng** [comment]
@harryyiheyang @avnostaeva ,

The test version is in the repository.    This verison will save all the beta in sparse matrix (only takes 100MB), and will save it out.

```
devtools::install_github("zhilizheng/SBayesRC")
```

 By default, it will have the BETA and SE in the output.txt (without any flags, this is the default)

Extract the SNP weight in first iteration by: 
```
SBayesRC::extractMCMCEff("YOURFOLDER/test", 1,  "weight_iter1.txt")
```
Then you can use it in the SBayesRC::prs. 

Let me know if you find any bugs. 

Regards,
Zhili

**Comment by zhilizheng** [comment]
@avnostaeva @harryyiheyang ,

Fixed in the version 0.2.6. 

Regards,
Zhili

**Comment by harryyiheyang** [comment]
Dear @zhilizheng ,

I am writing to express my sincere gratitude for your continuous patience and support in addressing the various questions and challenges I have encountered. Your willingness to assist has been invaluable to my work.

Thank you once again for your unwavering assistance and for creating such a remarkable piece of software.

Best regards,
Yihe
